### PR TITLE
added new id generator

### LINKED ISSUE #7: Update Payments
Feature Request

- [ ] The user receives a list with paid memberships, and the list contains a Member ID
- [ ] The user verifies that everything looks OK 
  * If everything looks OK, the user can accept, and the memberships are updated 
  * If something is not OK, the file should be moved to a backup file with a prefix like "error" + timestamp
  

Error handling should be handled manually by the "finance" department :)

### PR DESCRIPTION
**Pull request by Moshizzl3**
What kind of change does this PR introduce?

- [ ] `bug-fix`
- [x] `new-feature`
- [x] `enhancement`

### CODE REVIEW

**Review by Andreassim — APPROVED** [review]

File: src/main/java/com/app/controllers/CompetitionController.java [hunk 1]
```diff
@@ -192,4 +207,15 @@ public void saveCompetitionsToFile() {
...
+
+  private String generateID() {
+    int id;
+    try {
+      int temp = Integer.parseInt(competitions.get(competitions.size() - 1).getId());
+      id = temp + 1;
+    } catch (IndexOutOfBoundsException e) {
+      id = 1;
+    }
+    return Integer.toString(id);
```
Comment (MathiasReker) [inline src/main/java/com/app/controllers/CompetitionController.java:219]: String.valueOf()

File: src/main/java/com/app/controllers/MemberController.java [hunk 2]
```diff
@@ -328,7 +314,20 @@ public void saveMembers() {
...
+  private String generateID() {
+    int id;
+    try {
+      int temp = Integer.parseInt(members.get(members.size() - 1).getID());
+      id = temp + 1;
+    } catch (IndexOutOfBoundsException e) {
+      id = 1;
+    }
+
+    return Integer.toString(id);
```
Comment (MathiasReker) [inline src/main/java/com/app/controllers/MemberController.java:331]: String.valueOf()

**Review by Andreassim — APPROVED** [review]
LGTM

**Review by Jarkyman — APPROVED** [review]

File: src/main/java/com/app/models/services/PaymentService.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    }
+    fileService.writeToFile(result);
+  }
+
+  private String createNewPath() {
+    String s = path;
+    String[] data = s.split("/");
+    data[2] = "backup.txt";
+    s = String.join("/", data[0], data[1], LocalDate.now() + data[2]);
+    return s;
```
Comment (MathiasReker) [inline src/main/java/com/app/models/services/PaymentService.java]:
```suggestion
  private String createNewPath() {
    String[] data = path.split("/");
    data[2] = "backup.txt";
    return String.join("/", data[0], data[1], LocalDate.now() + data[2]);
```

Comment (Andreassim) [reply src/main/java/com/app/models/services/PaymentService.java]: dangerous to hardcode array indexes.

**Pull request merged**